### ISSUE TITLE
Strict config loading before overrides always fails

### ISSUE THREAD

**Issue by nathan-az** [NVIDIA-NeMo/RL#1060]
**Describe the bug**

Strict config loading in the policy worker models occurs before overrides are passed in from the user config. This means that if the defaults fail for whatever reason, but would work with the overrides, we see a failure.

**MRE**

<details>

```py
from megatron.bridge.training.config import ConfigContainer, LoggerConfig
from nemo_rl.models.policy.utils import (
    get_megatron_checkpoint_dir,
)
from nemo_rl.models.megatron.community_import import import_model_from_hf_name
import os
import torch

cfg = {
    "megatron_cfg": {
        "context_parallel_size": 1,
        "converter_type": "Qwen2ForCausalLM",
        "enabled": True,
        "expert_model_parallel_size": 4,
        "expert_tensor_parallel_size": 1,
        "num_layers_in_first_pipeline_stage": None,
        "num_layers_in_last_pipeline_stage": None,
        "pipeline_model_parallel_size": 1,
        "sequence_parallel": True,
        "tensor_model_parallel_size": 4,
    },
    "model_name": "Qwen/Qwen3-30B-A3B",
}
hf_model_name = cfg["model_name"]
hf_model_subdir = hf_model_name
if os.path.exists(hf_model_name):
    hf_model_subdir = f"model_{hf_model_subdir.replace('/', '_')}"

pretrained_path = f"{get_megatron_checkpoint_dir()}/{hf_model_subdir}"
pt_checkpoint_exists = os.path.exists(pretrained_path) and os.path.exists(
    os.path.join(pretrained_path, "iter_0000000")
)
torch.distributed.init_process_group("nccl")
if pt_checkpoint_exists:
    print(f"Checkpoint already exists at {pretrained_path}. Skipping import.")
else:
    import_model_from_hf_name( 
        hf_model_name, pretrained_path, cfg["megatron_cfg"]
    )

pretrained_run_config = os.path.join(
    pretrained_path, "iter_0000000/run_config.yaml"
)

from megatron.bridge.utils.instantiate_utils import InstantiationMode
cfg_from_pretrained = ConfigContainer.from_yaml(
    pretrained_run_config, mode=InstantiationMode.STRICT
)
model_cfg = cfg_from_pretrained.model
cfg_from_pretrained.logger = LoggerConfig()
```

</details>

```
[rank1]: megatron.bridge.utils.instantiate_utils.InstantiationException: Error in call to target 'megatron.bridge.models.qwen.qwen_provider.Qwen3MoEModelProvider':
[rank1]: ValueError('When using expert parallelism and tensor parallelism, sequence parallelism must be used')
[rank1]: full_key: model
```

Note that the MRE uses STRICT instantiation mode, because otherwise the error is masked and the `model` key becomes None then fails later in a more opaque way. https://github.com/NVIDIA-NeMo/RL/issues/1054 will merge this change anyway.

**Expected behavior**

If the user config has `sequence_parallel: true` it should not fail because the default is `false`.

**Environment overview (please complete the following information)**

 - Environment location: AWS
 - Method of install: `uv run python ...`
 - NVIDIA Nemo docker image (not NeMO-RL)

**Possible fixes**

1. add `sequence_parallel` override to [import_model_from_hf_name](https://github.com/NVIDIA-NeMo/RL/blob/13b4ca5a9b396dc5a8c82eee9c0c1c7df6c81736/nemo_rl/models/megatron/community_import.py#L57)
1. add more complex merge logic to the above to automatically merge the user's megatron config
1. in [MegatronPolicyWorker](https://github.com/NVIDIA-NeMo/RL/blob/13b4ca5a9b396dc5a8c82eee9c0c1c7df6c81736/nemo_rl/models/policy/megatron_policy_worker.py#L523), load `pretrained_run_config` as a dict first, then pass through all manual updates, then validate in strict mode using `from_dict` rather than `from_yaml`.

Happy to contribute here if there's a preferred path.

**Comment by qiaochuz-nv** [comment]
## QA RCCA Analysis

RCCA Status: Done
RCCA Category: missing_test_case
Test Action: create_new_test

Fix PR: https://github.com/NVIDIA-NeMo/RL/pull/1078
Root Cause: product-code fix PR found without upstream test file.
Trigger Config: Strict config loading before overrides always fails

Coverage Assessment:
| Area | Result |
|---|---|
| Existing Test | N/A |
| Coverage Status | MISSING_TEST_CASE |
| Gap | No direct upstream or nmfw_tests regression identified for this fixed code path. |
| Corrective Action | Missing regression test recorded. Add targeted nmfw_tests coverage if this issue recurs or remains high priority. |

Conclusion: RCCA classification completed for this closed NeMo-RL GitHub issue.
